api/api_ethereum.go

```diff
@@ -1222,41 +1220,28 @@ func (args *EthTransactionArgs) ToMessage(globalGasCap uint64, baseFee *big.Int,
 		logger.Warn("Caller gas above allowance, capping", "requested", gas, "cap", globalGasCap)
 		gas = globalGasCap
 	}
-	var (
-		gasPrice  *big.Int
-		gasFeeCap *big.Int
-		gasTipCap *big.Int
-	)
-	if baseFee == nil {
+
+	gasPrice := new(big.Int)
+	if baseFee.Cmp(new(big.Int).SetUint64(params.ZeroBaseFee)) == 0 {
```

Comment:
I think this cannot cover correctly when baseFee is nil.

Reply:
Right, but baseFee cannot be nil from the caller.

Reply:
Oh, okay i checked. Thanks